libc/src/setjmp/x86_64/longjmp.cpp

```diff
@@ -7,15 +7,34 @@
 //===----------------------------------------------------------------------===//
 
 #include "src/setjmp/longjmp.h"
+#include "include/llvm-libc-macros/offsetof-macro.h"
 #include "src/__support/common.h"
 #include "src/__support/macros/config.h"
 
-#if !defined(LIBC_TARGET_ARCH_IS_X86_64)
+#if !defined(LIBC_TARGET_ARCH_IS_X86)
 #error "Invalid file include"
 #endif
 
 namespace LIBC_NAMESPACE_DECL {
 
+#ifdef __i386__
+[[noreturn]]
+LLVM_LIBC_FUNCTION(void, longjmp, (jmp_buf buf, int val)) {
+  asm(R"(
+      mov %[ebx], %%ebx
+      mov %[esi], %%esi
+      mov %[edi], %%edi
+      mov %[ebp], %%ebp
+      mov %[esp], %%esp
+
+      jmp *%[eip]
```

Comment:
Fortunately/unfortunately, I could not reinterpret_cast `buf_eip` to a no-return function pointer and call it.